src/frontend/context/Kompetanse/valideringKompetanse.ts

```diff
@@ -0,0 +1,144 @@
+const erKompetanseResultatGyldig = (
+    felt: FeltState<KompetanseResultat | undefined>
+): FeltState<KompetanseResultat | undefined> =>
+    !isEmpty(felt.verdi) ? ok(felt) : feil(felt, ikkeUtfyltFelt);
+
+const erKompetansePeriodeGyldig = (
+    felt: FeltState<IYearMonthPeriode>,
+    avhengigheter?: Avhengigheter
+): FeltState<IYearMonthPeriode> => {
+    const fom = felt.verdi.fom;
+    const tom = felt.verdi.tom;
+
+    if (!fom || isEmpty(fom)) {
+        return feil(felt, 'Fra og med måned må være utfylt');
+    }
+    if (!erEtter(fom, avhengigheter?.initielFom)) {
+        return feil(
+            felt,
+            `Du kan ikke legge inn fra og med måned som er før: ${avhengigheter?.initielFom}`
+        );
+    }
+    if (tom && valgtTomErNesteMånedEllerSenere(tom)) {
+        return feil(
+            felt,
+            'Du kan ikke legge inn til og med måned som er i neste måned eller senere'
+        );
+    }
+
+    return ok(felt);
+};
+
+const validerKompetanse = (nyKompetanse: FeltState<IKompetanse>): FeltState<IKompetanse> => {
+    const nyVerdi: IKompetanse = {
+        ...nyKompetanse.verdi,
+        periode: nyKompetanse.verdi.periode.valider(nyKompetanse.verdi.periode, {
+            initielFom: nyKompetanse.verdi.initielFom,
+        }),
+        barnIdenter: nyKompetanse.verdi.barnIdenter.valider(nyKompetanse.verdi.barnIdenter),
+        søkersAktivitet: nyKompetanse.verdi.søkersAktivitet?.valider(
+            nyKompetanse.verdi.søkersAktivitet
+        ),
+        annenForeldersAktivitet: nyKompetanse.verdi.annenForeldersAktivitet?.valider(
+            nyKompetanse.verdi.annenForeldersAktivitet
+        ),
+        annenForeldersAktivitetsland: nyKompetanse.verdi.annenForeldersAktivitetsland?.valider(
+            nyKompetanse.verdi.annenForeldersAktivitetsland,
+            {
+                annenForeldersAktivitet: nyKompetanse.verdi.annenForeldersAktivitet,
+            }
+        ),
+        barnetsBostedsland: nyKompetanse.verdi.barnetsBostedsland?.valider(
+            nyKompetanse.verdi.barnetsBostedsland
+        ),
+        resultat: nyKompetanse.verdi.resultat?.valider(nyKompetanse.verdi.resultat),
+    };
+
+    const gyldigkompetanse: boolean =
+        nyVerdi.periode.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.barnIdenter.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.søkersAktivitet?.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.annenForeldersAktivitet?.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.annenForeldersAktivitetsland?.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.barnetsBostedsland?.valideringsstatus === Valideringsstatus.OK &&
+        nyVerdi.resultat?.valideringsstatus === Valideringsstatus.OK;
+
+    return gyldigkompetanse
+        ? ok({ ...nyKompetanse, verdi: nyVerdi })
+        : feil({ ...nyKompetanse, verdi: nyVerdi }, '');
+};
+
+export {
```

Comment:
Pleier vi å lage tester for valideringsregler? Det kunne jo ha vært nice, særlig for enkelte av de mer komplekse reglene

Reply:
Har ikkje funne tester for det.